src/NuGet.Clients/VisualStudio/Extensibility/CredentialRequestType.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// The type of credential request being made.
+    /// </summary>
+    public enum CredentialRequestType
```

Comment:
Ah, right -- Nuget always tries unauthenticated first, and there are no cases where we request credentials up front. I was thinking we'd have a state = 0, "None" or "Request", meaning no previous request had been made.